helix-core/src/test/java/org/apache/helix/controller/stages/TestRebalancePipeline.java

```diff
@@ -61,8 +62,11 @@ public void testDuplicateMsg() {
     HelixManager manager = new DummyClusterManager(clusterName, accessor);
     ClusterEvent event = new ClusterEvent(ClusterEventType.Unknown);
     event.addAttribute(AttributeName.helixmanager.name(), manager);
-    event.addAttribute(AttributeName.ControllerDataProvider.name(),
-        new ResourceControllerDataProvider());
+    ResourceControllerDataProvider dataCache = new ResourceControllerDataProvider();
+    // The AsyncTasksThreadPool needs to be set, otherwise to start pending message cleanup job
+    // will throw NPE and stop the pipeline.
```

Comment:
Once we create an issue, you could link that issue number in the comment so we could track it better.